service_crategen/src/commands/generate/codegen/rest_xml.rs

```diff
@@ -307,32 +314,32 @@ fn generate_list_serializer(shape: &Shape) -> String {
     };
 
     let element_type = &mutate_type_name(&shape.member_type()[..]);
-    let mut serializer = "let mut parts: Vec<String> = Vec::new();".to_owned();
+    let mut serializer = "".to_owned();
 
     if !flattened {
-        serializer += "parts.push(format!(\"<{}>\", name));";
+        serializer += "writer.write(xml::writer::XmlEvent::start_element(name))?;";
     }
 
     serializer += &format!("
         for element in obj {{
-            parts.push({element_type}Serializer::serialize(\"{element_type}\", element));
+            {element_type}Serializer::serialize(writer, \"{element_type}\", element)?;
         }}",
                            element_type = element_type);
 
     if !flattened {
-        serializer += "parts.push(format!(\"</{}>\", name));";
+        serializer += "writer.write(xml::writer::XmlEvent::end_element())?;";
     }
 
-    serializer += "parts.join(\"\")";
+    serializer += "Ok(())";
     serializer
 }
 
 fn generate_map_serializer(_shape: &Shape) -> String {
-    "String::new()".to_string()
+    unreachable!()
```

Comment:
Maybe we just want to remove this if it's unreachable?

Reply:
We use this in our matcher in rest_xml:

```rust
fn generate_serializer_body(shape: &Shape, service: &Service) -> String {
    match shape.shape_type {
        ShapeType::List => generate_list_serializer(shape),
        ShapeType::Map => generate_map_serializer(shape),
        ShapeType::Structure => generate_struct_serializer(shape, service),
        _ => generate_primitive_serializer(shape),
    }
}
```

A good way to change this is to document *why* we don't generate a map serializer for rest_xml services.  Which I don't know off the top of my head.  I suspect we don't because we just haven't implemented it yet.  Another reason may be rest_xml services don't use maps to serialize data.

I'm okay with adding a comment to the code with a `TODO: determine why we don't generate map serializers for rest_xml`.

That's the only change I'd make to this PR: tests run and pass. 🎉

Reply:
No rest-xml services use Map shape type, at least currently. I've added comment about that.
https://github.com/rusoto/rusoto/pull/812/commits/3e78792a753c09c2bf1fa52f50cc79d9647de4fb